src/NHibernate/Proxy/AbstractProxyFactory.cs

```diff
@@ -10,8 +11,9 @@ namespace NHibernate.Proxy
 	/// <summary>
 	/// Convenient common implementation for ProxyFactory
 	/// </summary>
-	public abstract class AbstractProxyFactory: IProxyFactory
+	public abstract class AbstractProxyFactory: IProxyFactory, ISerializable
```

Comment:
The easiest (and I think the _correct_) way to serialize a proxy was to serialize its proxy factory.

Reply:
Maybe...

Reply:
I now realize this point is a possible breaking change: static proxies are serialized without their entity state, even if they were initialized. They are always de-serialized in un-initialized state.

So, for proxies de-serialized by a session de-serialization, using them will always triggers a lazy load even if they were serialized while already being loaded. It looks to me dynamic proxies would not trigger a lazy load in that case.

For proxies directly serialized/de-serialized, with dynamic proxies their de-serialized state should be usable if they were initialized before serialization, while static proxies would be always un-initialized and un-usable because no more bound to a session.

Maybe a way to get the initializer `_target` instance should be added, for serializing it along the other proxy data, thus allowing to supply it back to the proxy on deserialization.